pyscriptjs/src/plugin.ts

```diff
@@ -89,6 +93,18 @@ export class PluginManager {
         for (const p of this._pythonPlugins) p.afterStartup?.(runtime);
     }
 
+    beforePyScriptExec(runtime, src, pyscriptTag) {
+        for (const p of this._plugins) p.beforePyScriptExec(runtime, src, pyscriptTag);
+
+        for (const p of this._pythonPlugins) p.beforePyScriptExec?.(runtime, src, pyscriptTag);
+    }
+
+    afterPyScriptExec(runtime: Runtime, src, pyscriptTag, result) {
+        for (const p of this._plugins) p.afterPyScriptExec(runtime, src, pyscriptTag, result);
+
+        for (const p of this._pythonPlugins) p.afterPyScriptExec?.(runtime, src, pyscriptTag, result);
+    }
+
```

Comment:
The first loop iterates over all the TypeScript/JavaScript plugins, the second iterates over all the Python plugins. It's pretty much the same behavior as all the other Plugin methods in that file.

The reason for using the optional chaining operator `?.` for the latter and not the former: 

- The JavaScript objects all inherit from [Plugin](https://github.com/JeffersGlass/pyscript/blob/pyscript-exec-hooks/pyscriptjs/src/plugin.ts), so they're guaranteed to have the required methods (`configure()`, `beforeLaunch()`,  `beforePyScriptExec()`) by inheritance if nothing else.
But since Python plugin classes don't have to inherit from a base class, the presence of those methods isn't guaranteed. 

The optional chaining operator `.?` is a short convenient handle this - `p.afterPyScriptExec?.(runtime, src...)` just means "If `p` has an attribute called `afterPyScriptExec`, call it with arguments `runtime`, `src`, etc. Otherwise, return undefined.

Reply:
Yeah! I just noticed is the default way people are doing plugins, I hadn't caught up on that yet.
Now I did. Thanks! :)